src/ray/rpc/client_call.h

```diff
@@ -155,8 +163,9 @@ class ClientCallManager {
       const Request &request, const ClientCallback<Reply> &callback) {
     auto call = std::make_shared<ClientCallImpl<Reply>>(callback);
     // Send request.
-    call->response_reader_ =
-        (stub.*prepare_async_function)(&call->context_, request, &cq_);
+    // Find a random completion queue to wait for response.
+    call->response_reader_ = (stub.*prepare_async_function)(&call->context_, request,
+                                                            &cqs_[rand() % num_threads_]);
```

Comment:
Do you know if it's better to bind all calls to the same client to the same completion queue if we are talking to many clients?

Reply:
I think this shouldn't be a problem. Completion queues are defined per server and client pair. gRPC is ultimately dispatching different rpcs to the set of completion queues.